src/pzt/istio/sections/20-operationalize/10-renewal/20-deployments/kubernetes/10-certificate.mdx

```diff
@@ -0,0 +1,94 @@
+```
+
+In cert-manager, a custom resource type `Certificate` represents
+each TLS certififcate and specifies how to store it.
+
+<p>
+  Create a new <inlineCode>Certificate</inlineCode> resource{' '}
+  <inlineCode>{props.pzt.server.name}-certificate.yaml</inlineCode> for your{' '}
+  {props.pzt.name} server. Make sure you specify the same DNS name configured
+  above, and keep track of the Kubernetes secret name where cert-manager will
+  store your certificate and private key.
+</p>
+
+<CodeBlock autoCopy language="yaml">
+  {`apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: ${props.pzt.server.name}
+spec:
+  secretName: ${props.pzt.server.name}-tls
+  duration: 24h
+  dnsNames:
+    - ${props.pzt.server.dnsName}
+  issuerRef:
+    name: my-ca
+    kind: ${props.page.provisioner === 'jwk' ? 'StepIssuer' : 'Issuer'}
+    group: ${
+      props.page.provisioner === 'jwk'
+        ? 'certmanager.step.sm'
+        : 'cert-manager.io'
+    }`}
+</CodeBlock>
+
+Customize the `Certificate` name, `secretName`, and issuer name to suit your own naming style in your Kubernetes cluster, then apply it with `kubectl`.
+
+<CodeBlock autoCopy language="shell">
+  {`kubectl apply -f ${props.pzt.server.name}-certificate.yaml`}
+</CodeBlock>
+
+<p>
+  This configuration specifies that cert-manager should issue and renew a TLS
+  certificate with the DNS name{' '}
+  <inlineCode>{props.pzt.server.dnsName}</inlineCode> and store the certificate
+  and private key in a Kubernetes secret named{' '}
+  <inlineCode>{props.pzt.server.name}-tls</inlineCode>. The certificate is valid
+  for 24 hours, and cert-manager will automatically renew it before expiration
+  and update the <inlineCode>{props.pzt.server.name}-tls</inlineCode> secret.
+</p>
+
+After creating the `Certificate` resource for cert-manager, we need to connect cert-manager to Istio.
+The easist way to connect is by using the <a href="https://github.com/cert-manager/istio-csr">istio-csr</a> agent.
+
+<p>
+  Change some of the Helm values before we deploy by creating{' '}
+  <inlineCode>{props.pzt.server.name}-istio-csr.yaml</inlineCode>. Make sure you
+  specify the same issuer name, kind, and group that was used in the{' '}
+  <inlineCode>Certificate</inlineCode> resource created above.
+</p>
+
+<CodeBlock autoCopy language="yaml">
+  {`certificate:
+ namespace: default
+ group: ${
+   props.page.provisioner === 'jwk' ? 'certmanager.step.sm' : 'cert-manager.io'
+ }
+ kind: ${props.page.provisioner === 'jwk' ? 'StepIssuer' : 'Issuer'}
+ name: my-ca
+ maxDuration: 24h
+ preserveCertificateRequests: false`}
+</CodeBlock>
+
+Next, install `istio-csr` in your cluster using Helm and the configuration file right above.
+
+<CodeBlock autoCopy language="shell">
+  {`helm install cert-manager-istio-csr jetstack/cert-manager-istio-csr -f ${props.pzt.server.name}-istio-csr.yaml --namespace cert-manager`}
+</CodeBlock>
```

Comment:
Same, this is a huge copy / paste with too much duplication. Maybe you can help me identify the unique parts and we can go from tehre. Or maybe this istio doc shouldn't be on `template: kubernetes` if they're too different?

Reply:
They have a weird overlap. The cert manager and step-issuer steps are necessary for istio too, like for the ingress controller. However, there needs to be an additional step to install the `istio-csr` to connect cert manager with istio. I feel like it should still use the Kubernetes template cause of the overlap for the first two parts, but let me comb through again to see how we can reduce the duplication issue.